src/main/java/org/checkerframework/specimin/UnsolvedClass.java

```diff
@@ -149,7 +165,11 @@ public void updateFieldByType(String currentType, String correctType) {
   public String toString() {
     StringBuilder sb = new StringBuilder();
     sb.append("package ").append(packageName).append(";\n");
-    sb.append("public class ").append(className).append(" {\n");
+    if (hasAPlaceHolder) {
```

Comment:
This logic doesn't seem to handle classes with more than one type variable, such as `java.util.Map`. Can you add a test case where an unsolved class has two different type variables?

Reply:
My other concern about this is that this code will never add an `extends` clause to the type variable, even if one is warranted. Maybe that's not possible if the class is unsolved, though.

Reply:
> My other concern about this is that this code will never add an extends clause to the type variable, even if one is warranted. Maybe that's not possible if the class is unsolved, though.

I've also thought about it.  I don't think we can figure out if an unsolved class should have an `extends` or `implements` clause.